File: wlog/src/main/java/me/shikhov/wlog/StringJuggler.java
package me.shikhov.wlog;


import androidx.annotation.NonNull;

class StringJuggler
{
    protected final StringBuilder stringBuilder;

    public StringJuggler(@NonNull StringBuilder stringBuilder)
    {
        this.stringBuilder = stringBuilder;
    }

    public boolean hasData()
    {
        return stringBuilder.length() > 0;
    }

    public String getString()
    {
        return stringBuilder.toString();
    }

    public void clear()
    {
        stringBuilder.delete(0, stringBuilder.length());
    }

    @NonNull
    public StringBuilder getStringBuilder()
    {
        return stringBuilder;
    }

    public StringJuggler append(String text)
    {
        stringBuilder.append(text);

        return this;
    }
}

File: wlog/src/main/java/me/shikhov/wlog/StringJuggler.kt
package me.shikhov.wlog

internal class StringJuggler(val stringBuilder: StringBuilder,
                             var separator: CharSequence?) : Appendable {

    fun isNotEmpty() = stringBuilder.isNotEmpty()

    val string: String
        get() = stringBuilder.toString()

    fun clear() {
        stringBuilder.clear()
    }

    override fun append(csq: CharSequence?): java.lang.Appendable {
        stringBuilder.append(csq)
        separator?.let { stringBuilder.append(it) }
        return this
    }

    override fun append(csq: CharSequence?, start: Int, end: Int): java.lang.Appendable {
        stringBuilder.append(csq, start, end)
        separator?.let { stringBuilder.append(it) }
        return this
    }

    override fun append(c: Char): java.lang.Appendable {
        stringBuilder.append(c)
        separator?.let { stringBuilder.append(it) }
        return this
    }
}